### ISSUE TITLE
v.2.0.64: Settings are not saved

### ISSUE THREAD

**Issue by Speravir** [aaFn/Bookmark-search-plus-2#111]
On updating to Firefox 68 I noticed that your addon does not work: It tries unsuccessfully to load the bookmarks. I looked into the preferences and noticed some things:
* My setting for advanced click behaviour was set to the default. I switched it back, but this is not saved.
* There are some performance settings that I attempted to activate to old behaviour, but these are not saved, too.
* After manually loading the bookmarks by clicking on the according button the addon works temporarily even without setting to old behaviour, but after a restart this is gone.

Strange enough my change for the key shortcut has been saved.

**Comment by aaFn** [comment]
Hello @Speravir , sorry I was traveling.

Well, I just checked on my side, this continues to work well on option settings = they are saved and kept across FF stop / restart.
That said, I was just upgraded to 68.0.1 .. don't know if that changes anything ?

If you confirm you still have the problem with 68.0.1, coud I ask that you:
- open the browser console (Ctrl+Shit+J)
- empty it (top left trash icon)
- then go to the BSP2 option page (preferences), and change a setting which is usually not saved for you
- look at the browser console, and copy its contents (if any ...) here

Hoping there would be an exception or error message showing ..

Similarly, if you can;
- Activate the trace option
- Copy here the contents of the trace window which should appear at top of the bookmark tree in the sidebar

Can you also clarify "tries unsuccessfully to load the bookmarks" ? Do you mean that nothing is dipsplayed and you have the spinning wheel stuck on a step ? If so, what is the displayed step message ?
Or just an empty, or nearly empty, tree ? What do the statistics show in the option page after pressing the "Reload from FF" button ? (refresh the option page after doing so, to make sure you have the latest numbers)

All that you say point to some form of corruption of BSP2 local storage.
I must say that since FF66 and their change of local storage to use MySQL instead of simple json files, there are plenty of unstabilities and performance impacts on this side .. and this is harder to investigate, because I couldn't figure out where this is saved now, and how to verify what is inside, so cannot really debug with users when they have trouble .. :-(

Anyway, let's see what you get. At worst, you should uninstall BSP2 and resintall it, that will reset its local storage .. but doing so you will lose the favicons and the process to gather them will start again, unless to check (activate) the corresponding Pause or Disable function (pause is better, as it does not forget what you fetched so far). Depending on the number of bookmarks you have, if not too big, that could be a good strategy in case of corruption.

**Comment by Speravir** [comment]
TLDR: The addons works, again. It seems the local storage was totally corrupt. Sorry for the noise.

I started the browser console and saw … nothing, literally nothing, or: an empty log.

Then I remembered that I had issues with some other addons after updating to Fx68. In these cases I could solve this with reopening a backup of my profile, starting this with Fx67, then exporting the settings (these other addons provide this) and afterwards importing in Fx68. But for one of the addons (Stylus) I had taken some research before and there were suggestions related to local storage. So I thought let me check whether this could be the reason also for BMS++.

With `about:debugging` I found the internal UUID of your addon, and then I could search for the addon folder in local storage, `moz-extension+++$UUID$^userContextId=$numberstring$`. I deleted this folder, and after a restart of Firefox your addon now works fine. I had to readjust everything and the favicons are gone, but I can live with this.

**Comment by aaFn** [comment]
Hello @Speravir , ok thank you much for feedback and for the details, that allows me to better understand where is the data stored now ... I guess I should find a sqlite editor to open them ..

Well, that is worrying a bit though, that confirms corruptions as I suspected can happen now, but no hint of what causes them ... and no easy way to recover (except from uninstalling and reinstalling, I am afraid) :-(

Let me know if that happens again.

------------------------------------------------

Now on another topic, I guess you know this guy https://github.com/Aris-t2/CustomCSSforFx
I see he made some userChrome.css mods specific to BSP2, I can look at how to ease things, if needed and if he has suggestions,

**Comment by Speravir** [comment]
>  I guess I should find a sqlite editor to open them ..

The contents are encrypted, though. Probably the best would be to use the internal library/-ies. BTW there are even addons for sqlite. I am not a pro at all, but this program seems quite good to me (I use for opening sqlite databases): [DB Browser for SQLite](https://sqlitebrowser.org/).

> Well, that is worrying a bit though, that confirms corruptions as I suspected can happen now, but no hint of what causes them ... and no easy way to recover (except from uninstalling and reinstalling, I am afraid) :-(

Well, as I said your addon wasn’t the only one. But the others have exporting and importing routines. I do not know whether something like this would help here. What to export and then import? As your addon worked again the favicons have been restored quite fast.

> Now on another topic, I guess you know this guy https://github.com/Aris-t2/CustomCSSforFx
> I see he made some userChrome.css mods specific to BSP2, I can look at how to ease things, if needed and if he has suggestions,

Haha, guess once, whom he answered [here](https://github.com/Aris-t2/CustomCSSforFx/issues/109#issuecomment-402849175) :-) Actually I can tell you that I use this user style since 9 March 2018 (first version by Aris). Let’s discuss this in an own issue ticket.
